services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_scheduler.py

```diff
@@ -248,6 +249,34 @@ async def mark_service_for_removal(
             current.dynamic_sidecar.service_removal_state.mark_to_remove(can_save)
             await update_scheduler_data_label(current)
 
+            # cancel current observation task
+            if service_name in self._service_observation_task:
+                service_task: Optional[
+                    Union[asyncio.Task, object]
```

Comment:
you have three states possible here: asyncio.Task, None and ... object? Is this some sort of sentinel object? Can you perhaps make it more explicit. For instance, this is a pattern i like
https://github.com/python/cpython/blob/3.10/Lib/dataclasses.py#L182-L186

Reply:
I think this is specific enough for what needs to be done here. If it is an `asyncio.Task` it needs to be cancelled.